lib/index.ts

```diff
@@ -29,31 +29,42 @@ export const generateRemoteUrl = (service: string) => window.location.protocol +
 /**
  * Get the base path for the given OCS API service
  *
- * @param {string} service name
- * @param {int} version OCS API version
- * @return {string} OCS API base path
+ * @param {string} url OCS API service url
+ * @param {object} params parameters to be replaced into the service url
+ * @param {UrlOptions} options options for the parameter replacement
+ * @param {boolean} options.escape Set to false if parameters should not be URL encoded (default true)
+ * @param {Number} options.ocsVersion OCS version to use (defaults to 2)
+ * @return {string} Absolute path for the OCS URL
  */
-export const generateOcsUrl = (service: string, version: Number) => {
-    version = (version !== 2) ? 1 : 2
-    return window.location.protocol + '//' + window.location.host + getRootUrl() + '/ocs/v' + version + '.php/' + service + '/'
+export const generateOcsUrl = (url: string, params?: object, options?: UrlOptions) => {
+    const allOptions = Object.assign({
+        ocsVersion: 2
+    }, options || {})
+
+   const version = (allOptions.ocsVersion === 1) ? 1 : 2
```

Comment:
in theory we can now just use `allOptions.ocsVersion` as it's documented to be a number. If anyone specifies otherwise it's their fault if the function returns an invalid URL